src/Spryker/Zed/EventBehavior/Business/Model/EventResourceQueryContainerPluginIterator.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+/**
+ * Copyright © 2016-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace Spryker\Zed\EventBehavior\Business\Model;
+
+use Iterator;
+use Propel\Runtime\ActiveQuery\ModelCriteria;
+use Spryker\Zed\EventBehavior\Dependency\Plugin\EventResourceQueryContainerPluginInterface;
+
+class EventResourceQueryContainerPluginIterator implements Iterator
+{
+    /**
+     * @var \Spryker\Zed\EventBehavior\Dependency\Plugin\EventResourceQueryContainerPluginInterface
+     */
+    protected $plugin;
+
+    /**
+     * @var int
+     */
+    protected $index = 0;
+
+    /**
+     * @var array
+     */
+    protected $current = [];
+
+    /**
+     * @var int
+     */
+    protected $chunkSize;
```

Comment:
you can provide a default chunk size here e.g `1000`

Reply:
It's a required parameter in the constructor.

Reply:
Added to the environment configs.